cfme/tests/configure/test_tag.py

```diff
@@ -230,10 +238,19 @@ def test_create_tag_with_wrong_arguments(self, appliance):
         assert_response(appliance, http_status=400)
 
     @pytest.mark.tier(3)
-    @pytest.mark.parametrize(
-        "collection_name", ["clusters", "hosts", "data_stores", "providers", "resource_pools",
-        "services", "service_templates", "tenants", "vms", "availability_zones", "cloud_networks",
-        "cloud_networks", "cloud_subnets", "flavors", "network_routers", "security_groups"])
+    @pytest.mark.provider(
+        [CloudProvider, InfraProvider], selector=ONE_PER_CATEGORY, override=True
+    )
+    @pytest.mark.parametrize("collection_name", INFRA_COLLECTION + CLOUD_COLLECTION)
+    @pytest.mark.uncollectif(
+        lambda appliance, collection_name, provider: (
+            provider.one_of(CloudProvider) and collection_name in INFRA_COLLECTION
+        )
+        or (
+            (provider.one_of(InfraProvider) or appliance.version < "5.10")
+            and collection_name in CLOUD_COLLECTION
```

Comment:
Just checking my understanding of this. So here, you are uncollecting if the provider is a `CloudProvider` and the `collection` is related to an Infra provider, OR if the provider is an `InfraProvider` and the `collection` is related to a Cloud provider OR if the version of the appliance is "5.9" and the `collection` is related to a Cloud provider. 

Is there a cleaner way to do this, so that the tests aren't collected in the first place? I think the version uncollection is okay, but their may be a way to avoid the other uncollections.

Reply:
Um, I could write separate test cases for both the providers, but that's not the best option. There's one more way - using `pytest_generate_tests(metafunc)` to parametrize test, but afaik we're avoiding it.